flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -495,6 +520,11 @@ public FlinkKafkaProducer(
 		this.schema = checkNotNull(serializationSchema, "serializationSchema is null");
 		this.producerConfig = checkNotNull(producerConfig, "producerConfig is null");
 		this.flinkKafkaPartitioner = checkNotNull(customPartitioner, "customPartitioner is null").orElse(null);
+		if (customPartitioner.isPresent()) {
```

Comment:
Hmm, not sure if `KafkaSerializationSchema` can fully replace the partitioner. The current API does not provide number of partitions in `KafkaSerializationSchema.serialize()`, right? So the serializer will not be able to know the total number of partitions. Should we add that to the method arguments?